cpp/tests/prims/mg_count_if_v.cu

```diff
@@ -0,0 +1,178 @@
+struct test_predicate {
+  int mod;
+  test_predicate(int mod_count) : mod(mod_count) {}
+  __device__
+  bool operator()(const vertex_t &val) {
+    cuco::detail::MurmurHash3_32<vertex_t> hash_func{};
+    return (0 == (hash_func(val) % mod));
+  }
+};
+
+template <typename input_usecase_t>
+class Tests_MG_CountIfV
+  : public ::testing::TestWithParam<input_usecase_t> {
+ public:
+  Tests_MG_CountIfV() {}
+  static void SetupTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  // Compare the results of count_if_v primitive and thrust count_if on a single GPU
+  template <typename vertex_t, typename edge_t, typename weight_t, typename result_t>
+  void run_current_test(input_usecase_t const &input_usecase)
+  {
+    // 1. initialize handle
+
+    raft::handle_t handle{};
+    HighResClock hr_clock{};
+
+    raft::comms::initialize_mpi_comms(&handle, MPI_COMM_WORLD);
+    auto &comm           = handle.get_comms();
+    auto const comm_size = comm.get_size();
+    auto const comm_rank = comm.get_rank();
+
+    auto row_comm_size = static_cast<int>(sqrt(static_cast<double>(comm_size)));
+    while (comm_size % row_comm_size != 0) { --row_comm_size; }
+    cugraph::partition_2d::subcomm_factory_t<cugraph::partition_2d::key_naming_t, vertex_t>
+      subcomm_factory(handle, row_comm_size);
+
+    // 2. create MG graph
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      hr_clock.start();
+    }
+    auto [mg_graph, d_mg_renumber_map_labels] =
+      input_usecase.template construct_graph<vertex_t, edge_t, weight_t, true, true>(
+        handle, true, true);
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG construct_graph took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    auto mg_graph_view = mg_graph.view();
+
+    const int hash_bin_count = 5;
+
+    // 4. run MG count if
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      hr_clock.start();
+    }
+
+    vertex_t const * data = (*d_mg_renumber_map_labels).data();
+    auto vertex_count = count_if_v(handle,
+                                   mg_graph_view,
+                                   data,
+                                   test_predicate<vertex_t>(hash_bin_count));
+
+    if (PERF) {
+      CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      handle.get_comms().barrier();
+      double elapsed_time{0.0};
+      hr_clock.stop(&elapsed_time);
+      std::cout << "MG count if took " << elapsed_time * 1e-6 << " s.\n";
+    }
+
+    // 5. compare SG & MG results
+    auto [sg_graph, d_sg_renumber_map_labels] =
```

Comment:
```d_sg_renumber_map_labels``` is unused as the input argument renumber is set to false in calling construct_graph.

We can do
```
cugraph::experimental::graph_t<vertex_t, edge_t, weight_t, true, false> sg_graph(handle);
std::tie(sg_graph, std::ignore) =
          input_usecase.template construct_graph<...
```
to avoid defining an unused variable.